public/pages/Indices/utils/helpers.ts

```diff
@@ -20,3 +27,184 @@ export function getURLQueryParams(location: { search: string }): IndicesQueryPar
+  } else {
+    const errorMessage = `There is a problem set index setting for ${props.indexName}, please check with Admin`;
+    props.coreServices.notifications.toasts.addDanger(result?.error || errorMessage);
+    throw new Error(result?.error || errorMessage);
+  }
+}
+
+export async function getAlias(props: { aliasName?: string; commonService: CommonService }) {
+  return await props.commonService.apiCaller<{ alias: string }[]>({
+    endpoint: "cat.aliases",
+    method: "GET",
+    data: {
+      format: "json",
+      name: `${props.aliasName || ""}*`,
+      expand_wildcards: "open",
+    },
+  });
+}
+
+export async function splitIndex(props: {
+  sourceIndex: String;
+  targetIndex: String;
+  settingsPayload: Required<IndexItem>["settings"];
+  commonService: CommonService;
+  coreServices: CoreStart;
+}) {
+  const { aliases, ...settings } = props.settingsPayload;
+  const result = await props.commonService.apiCaller({
+    endpoint: "indices.split",
+    method: "PUT",
+    data: {
+      index: props.sourceIndex,
+      target: props.targetIndex,
+      body: {
+        settings: {
+          ...settings,
+        },
+        aliases,
+      },
+    },
+  });
+
+  if (result && result.ok) {
+    const toastInstance = props.coreServices.notifications.toasts.addSuccess(
+      `Successfully started splitting ${props.sourceIndex}. The split index will be named ${props.targetIndex}`,
+      {
+        toastLifeTimeMs: 1000 * 60 * 60 * 24 * 5,
+      }
+    );
+    await jobSchedulerInstance.addJob({
+      interval: 30000,
+      extras: {
+        toastId: toastInstance.id,
+        sourceIndex: props.sourceIndex,
+        destIndex: props.targetIndex,
+      },
+      type: "split",
+    } as RecoveryJobMetaData);
+    return result;
+  } else {
+    const errorMessage = `There was a problem submit split index request, please check with admin`;
+    props.coreServices.notifications.toasts.addDanger(result?.error || errorMessage);
+    throw new Error(result?.error || errorMessage);
+  }
+}
+
+/*
+ ** When splitting an index, the valid value of number_of_shards in the target index depends on the source index’s primary shards count:
+ ** (1) If the source index’s primary shards count is equal to 1,
+ **     then the valid value is an arbitrary number between 1 and 1024(includes 1 and 1024).
+ ** (2) If the source index’s primary shards count is greater than 1,
+ **     the valid value is obtained by continuously multiplying the source index’s primary shards count by 2
+ **     as long as it's not larger than 1024.
+ **     For example, if the source index has 3 primary shards, then all the valid value are 3,6,12,24,…,768.
+ */
+export function getSplitShardOptions(sourceShards: number) {
+  const shardsSelectOptions = [];
+  if (sourceShards == 1) {
+    for (let i = 2; i <= 1024; i++) {
+      shardsSelectOptions.push({
+        label: i.toString(),
+      });
+    }
+  } else {
+    const SHARDS_HARD_LIMIT = 1024 / 2;
```

Comment:
What is the significance of using `1024 / 2` instead of `512` directly?

Reply:
I have updated the code to define 1024 as a constant variable MAX_SHARDS_NUMBER.  I prefer to use MAX_SHARDS_NUMBER/2 so whoever look at the code knows where the limit is from. 512 is less straightforward.